### DISCUSSION TITLE
What is the correct way to cancel a task from within the task handler?

### DISCUSSION THREAD

**Discussion by strobus** [hibiken/asynq#461] [Q&A] [answered]
Certain conditions may be present in an application that would make it unable to continue with a task. Retries would be futile. In this case I need to cancel the task from within the handler. Is there a proper way to do this within the Asynq API?

**Comment by hibiken** [comment, marked as answer]
Would [`SkipRetry`](https://pkg.go.dev/github.com/hibiken/asynq#SkipRetry) work for your use case?
See https://github.com/hibiken/asynq/wiki/Task-Retry#skip-retry for an example.
LMK if that works!

**Reply by strobus** [reply]
Conceptually, I am looking for the equivalent of a 'dead letter queue'. I just read the Life of a Task entry on the wiki. I did not realize that the 'Archived' state was meant for this. So yes I think this will work fine. Many thanks!

**Reply by hibiken** [reply]
@strobus Just curious; what did you think "archived" state was meant for before reading the wiki? I could revisit the naming of states :)
